cluster/examples/kubernetes/cockroachdb/operator.yaml

```diff
@@ -0,0 +1,106 @@
+apiVersion: v1
+kind: Namespace
+metadata:
+  name: rook-cockroachdb-system
+---
+apiVersion: apiextensions.k8s.io/v1beta1
+kind: CustomResourceDefinition
+metadata:
+  name: clusters.cockroachdb.rook.io
+spec:
+  group: cockroachdb.rook.io
+  names:
+    kind: Cluster
+    listKind: ClusterList
+    plural: clusters
+    singular: cluster
+  scope: Namespaced
+  version: v1alpha1
+---
+apiVersion: rbac.authorization.k8s.io/v1beta1
+kind: ClusterRole
+metadata:
+  name: rook-cockroachdb-operator
+rules:
+- apiGroups:
+  - ""
+  resources:
+  - namespaces
+  - secrets
```

Comment:
does the operator need all these privs, including secrets?

Reply:
I swear I implemented the operator with zero permissions at first and then just kept iterating adding one by one until the operator could complete successfully.  but looking at some of these i'm questioning if my mind remembers the truth!  maybe for the core api group i was a bit permissive, but the other groups (e.g. apps, policy) i added as needed.  I'll reexamine the core API permissions.